src/main/java/core/basesyntax/strategy/operation/BalanceOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.FruitTransaction;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        addNewFruitBalance(transaction.getFruit(),
+                removeOldFruitBalance(transaction.getFruit()) + transaction.getQuantity());
+    }
+
+    private Integer removeOldFruitBalance(Fruit fruit) {
+        Integer oldFruitBalance = FruitStorage.fruitsMap.remove(fruit);
+        return oldFruitBalance == null ? 0 : oldFruitBalance;
+    }
```

Comment:
```suggestion
    public void handle(FruitTransaction transaction) {
        addNewFruitBalance(transaction.getFruit(),
                getOldFruitBalance(transaction.getFruit()) + transaction.getQuantity());
    }

    private Integer getOldFruitBalance(Fruit fruit) {
        Integer oldFruitBalance = FruitStorage.fruitsMap.get(fruit);
        return oldFruitBalance == null ? 0 : oldFruitBalance;
    }
```
There is no need to remove anything